src/pokemon.c

```diff
@@ -3424,13 +3435,13 @@ u32 GetBoxMonData(struct BoxPokemon *boxMon, s32 field, u8 *data)
         retVal = boxMon->language;
         break;
     case MON_DATA_SANITY_IS_BAD_EGG:
-        retVal = boxMon->isBadEgg;
+        retVal = IsBoxMonBadEgg(boxMon, field);
         break;
     case MON_DATA_SANITY_HAS_SPECIES:
         retVal = boxMon->hasSpecies;
         break;
     case MON_DATA_SANITY_IS_EGG:
-        retVal = boxMon->isEgg;
+        retVal = boxMon->isEgg | IsBoxMonBadEgg(boxMon, field);
```

Comment:
|| instead of |

Reply:
I don't think that's correct? `||` is a logical OR used for comparisons. Here a bitwise OR is being used to decide if `retVal` should be set to either 0 or 1.

Reply:
It's intentional, either can be only 0 or 1, and if one of any of these are 1, retVal should be one.
So if the result of boxMon->isEgg is in r0 and the result of IsBoxMonBadEgg is in r1, I would do
orr r0, r1
And then return r0, which is what this part of the code does

Reply:
My bad.